File: discount_biz/lib/geocoding.js
// lib/geocoding.js
const GOOGLE_MAPS_API_KEY = process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY;

export class GeocodingService {
  static async geocodeAddress(address) {
    try {
      const response = await fetch(
        `https://maps.googleapis.com/maps/api/geocode/json?address=${encodeURIComponent(address)}&key=${GOOGLE_MAPS_API_KEY}`
      );
      const data = await response.json();
      
      if (data.status === 'OK' && data.results.length > 0) {
        const result = data.results[0];
        return {
          success: true,
          latitude: result.geometry.location.lat,
          longitude: result.geometry.location.lng,
          formatted_address: result.formatted_address,
          place_id: result.place_id,
          address_components: result.address_components
        };
      }
      
      return {
        success: false,
        error: 'Address not found'
      };
    } catch (error) {
      return {
        success: false,
        error: 'Geocoding service unavailable'
      };
    }
  }
}

File: discount_api/app/utils/geocoding.py
# Create app/utils/geocoding.py
import aiohttp
import os
from typing import Optional, Dict, Any

async def geocode_address(address: str) -> Optional[Dict[str, Any]]:
    """
    Geocode an address using Google Geocoding API
    """
    api_key = os.getenv("GOOGLE_MAPS_API_KEY")
    if not api_key:
        return None
    
    url = "https://maps.googleapis.com/maps/api/geocode/json"
    params = {
        "address": address,
        "key": api_key
    }
    
    try:
        async with aiohttp.ClientSession() as session:
            async with session.get(url, params=params) as response:
                if response.status == 200:
                    data = await response.json()
                    
                    if data["status"] == "OK" and data["results"]:
                        result = data["results"][0]
                        location = result["geometry"]["location"]
                        
                        return {
                            "latitude": location["lat"],
                            "longitude": location["lng"],
                            "formatted_address": result["formatted_address"],
                            "place_id": result.get("place_id"),
                            "address_components": result.get("address_components")
                        }
                return None
    except Exception as e:
        print(f"Geocoding error: {e}")
        return None